src/xrpld/app/tx/detail/DeleteAccount.cpp

```diff
@@ -203,12 +222,10 @@ DeleteAccount::preclaim(PreclaimContext const& ctx)
         return tecDST_TAG_NEEDED;
 
     // Check whether the destination account requires deposit authorization.
-    if (ctx.view.rules().enabled(featureDepositAuth) &&
-        (sleDst->getFlags() & lsfDepositAuth))
-    {
-        if (!ctx.view.exists(keylet::depositPreauth(dst, account)))
-            return tecNO_PERMISSION;
-    }
+    auto const err = DepositPreauth::preclaimCheckCredentials(
```

Comment:
The Payment transaction was reverted to `if (!credentialsEnabled) {} else {}`  as it has different logic from other. Could you please check the latest version of these function. For example `preauthPreclaimCheck` - 20 rows of amendments check and 10 rows of fields check. I'm not sure about copy  / paste it everywhere. Just to confirm - if you still think that it will be better, I'll remove these helpers and put code directly into transactions.